### ISSUE TITLE
[docdb] Get rid of the liveness info column in YSQL whenever possible

### ISSUE THREAD

**Issue by mbautin** [yugabyte/yugabyte-db#4364]
Jira Link: [DB-1799](https://yugabyte.atlassian.net/browse/DB-1799)
We store a special column called a "liveness info column" to indicate row presence in case all columns are NULL, because we don't store NULLs at all. In YSQL where there are no TTLs and no special INSERT/UPDATE semantics of row creation that exists in YCQL, we don't need this column in many cases. We certainly don't need it if there is at least one non-nullable non-primary-key column. We need to be careful on schema changes, when all non-primary key columns become nullable. In such cases, when rows are inserted or updated in a way that all columns might become NULL, we need to start writing the liveness info column again.

Another possibility is to replace the liveness info column with the top-of-the-row record, with a DocKey without a subkey (similar to the full row overwrite or deletion record) in YSQL.

**Comment by kriteshranjan** [comment]
AI validation notes (`yb-issue-4364-analysis.md`):

# Analysis: yugabyte/yugabyte-db#4364 (Jira DB-1799)

Issue: "[docdb] Get rid of the liveness info column in YSQL whenever possible"
(opened 2020-05-01, kind/enhancement, area/docdb, priority/medium, still OPEN, zero comments).
Validation date: 2026-09-01. Kind: enhancement / feature-gap — behavior-delta validation
(current vs requested), not a crash repro.

## Verdict
NOT-REPRODUCED — the storage inefficiency this enhancement targets (a per-row liveness
system column written for every YSQL insert) does not exist in the default configuration
of any current in-support release: the packed-row feature (default-on for new universes
since v2.20.0) replaced the liveness column with exactly the "top-of-the-row record with
a DocKey without a subkey" that the issue itself proposed as the alternative design.
Quick-side status: Fixed (delivered by packed rows). The narrower option 1 in the issue
(skip liveness when a non-nullable non-PK column exists) was never implemented, but it
only matters on the non-default `ysql_enable_packed_row=false` path.

## Root cause
Current-state mechanism, master HEAD 77ce2ba47bd2b84265a7f65650f476496f7056fd
(2026-09-01T13:19:13Z):

- `src/yb/docdb/pgsql_operation.cc:1528` `PgsqlWriteOperation::ApplyInsert` decides per
  insert via `ShouldYsqlPackRow` (`:167-170`, gated by `FLAGS_ysql_enable_packed_row`
  and, for colocated tables, `FLAGS_ysql_enable_packed_row_for_colocated_table`).
- Packed path: `RowPackContext::Complete` (`src/yb/docdb/pgsql_operation.cc:1217-1224`)
  writes the whole row as ONE record at `DocPath(encoded_doc_key)` — the row DocKey with
  no subkey. No liveness column is written anywhere in this path (`:1607-1632`). This is
  verbatim the issue's alternative proposal.
- Unpacked fallback (`:1633-1637`): writes `KeyEntryValue::kLivenessColumn`
  (`SystemColumnId(0)`, defined `src/yb/dockv/key_entry_value.h:46`) with value
  `kNullLow` UNCONDITIONALLY — the nullability-conditional skip (issue option 1) was
  never implemented. Same unconditional write in the pggate-packed-request sub-path when
  packing is off (`:1588-1591`).
- Flag default: `DEFINE_RUNTIME_AUTO_bool(ysql_enable_packed_row, kExternal,
  !kYsqlEnablePackedRowTargetVal, kYsqlEnablePackedRowTargetVal)` with target
  `!yb::kIsDebug` = true in release builds (`src/yb/docdb/pgsql_operation.cc:115-119`).
  2024.2 uses auto-flag class `kNewInstallsOnly` `(false, true)` (its
  `pgsql_operation.cc:111` at v2024.2.11.0). Either way a new cluster starts with the
  flag promoted to true (observed `Auto true` on all five images).
- Compactions also retro-fit the design: when repacking is allowed, a legacy liveness KV
  triggers packing and is itself dropped from the output
  (`src/yb/docdb/docdb_compaction_context.cc:1342-1372` at local rev 46a9f7adfd).

Residual gap (why the issue is not 100% moot): with `ysql_enable_packed_row=false`
(operator opt-out; or a pre-v2.20-created universe whose auto flags were never
promoted), every YSQL insert still writes the liveness KV even when the table has a
non-nullable non-PK column.

## Trigger matrix
Predicted from source before any run; observed on all five resolved images
(new single-node cluster per image, `yugabyted start`).

| condition | predicted | observed |
|---|---|---|
| INSERT into `t_pack (k INT PRIMARY KEY, v INT NOT NULL)`, default flags | 1 record `SubDocKey(DocKey(...), [HT{...}]) -> PACKED_ROW...`; zero `SystemColumnId(0)` | confirmed 5/5 targets |
| INSERT into `t_nopack` (same shape) after `yb-ts-cli set_flag -force ysql_enable_packed_row false` | `SubDocKey(..., [SystemColumnId(0); HT{...}]) -> null` + `[ColumnId(1); ...] -> 42`, liveness written despite NOT NULL column | confirmed 5/5 targets |
| flag value pre/post flip | `Auto true` -> `Custom false` (runtime-settable) | confirmed 5/5 targets |

(2025.2/2026.1 sst_dump labels the record `PACKED_ROW_V1[0]`, 2024.2/2025.1/2.25.2 label
it `PACKED_ROW[0]`; identical payload `05000000480000002A` — formatter naming only.)

## Reproduction
Script: `/tmp/claude/yb4364-repro.sh` (per-target logs consolidated in
`/tmp/claude/yb4364-matrix.log`). Shape per image:

```sh
IMG=docker.io/yugabytedb/yugabyte:2026.1.1.1-b2-aarch64   # one of the 5 resolved tags
podman run -d --name yb4364 $IMG bash -c \
  '/home/yugabyte/bin/yugabyted start --advertise_address 127.0.0.1 --base_dir=/tmp/ybd --daemon=false'
# wait for YSQL, then:
podman exec yb4364 bash -c '/home/yugabyte/bin/ysqlsh -h 127.0.0.1 -U yugabyte -d yugabyte \
  -c "CREATE TABLE t_pack   (k INT PRIMARY KEY, v INT NOT NULL) SPLIT INTO 1 TABLETS" \
  -c "CREATE TABLE t_nopack (k INT PRIMARY KEY, v INT NOT NULL) SPLIT INTO 1 TABLETS" \
  -c "INSERT INTO t_pack VALUES (1, 42)"'
podman exec yb4364 bash -c '/home/yugabyte/bin/yb-ts-cli --server_address=127.0.0.1:9100 \
  set_flag -force ysql_enable_packed_row false'
podman exec yb4364 bash -c '/home/yugabyte/bin/ysqlsh -h 127.0.0.1 -U yugabyte -d yugabyte \
  -c "INSERT INTO t_nopack VALUES (1, 42)"'
podman exec yb4364 bash -c '/home/yugabyte/bin/yb-admin --master_addresses=127.0.0.1:7100 \
  flush_table ysql.yugabyte t_pack 60 && /home/yugabyte/bin/yb-admin \
  --master_addresses=127.0.0.1:7100 flush_table ysql.yugabyte t_nopack 60'
# locate each table's tablet dir under /tmp/ybd/data/yb-data/tserver/data/rocksdb/table-<id>/
# and dump:
podman exec yb4364 bash -c '/home/yugabyte/bin/sst_dump --command=scan \
  --file=<tabletdir>/000010.sst --output_format=decoded_regulardb'
```

Success signature (delivered behavior, default flags — `t_pack`):
```
SubDocKey(DocKey(0x1210, [1], []), [HT{ physical: ... }]) -> PACKED_ROW_V1[0](05000000480000002A)
```
Control / legacy behavior (`ysql_enable_packed_row=false` — `t_nopack`):
```
SubDocKey(DocKey(0x1210, [1], []), [SystemColumnId(0); HT{ physical: ... }]) -> null
SubDocKey(DocKey(0x1210, [1], []), [ColumnId(1); HT{ physical: ... w: 1 }]) -> 42
```
Grep test: `SystemColumnId(0)` absent in the t_pack dump, present in the t_nopack dump.

## Versions tested
Resolved at: 2026-09-01T14:12:31Z (controller); independently re-verified
2026-09-01T14:19:39Z (this session).
Resolution evidence: Docker Hub tags API (`hub.docker.com/v2/repositories/yugabytedb/yugabyte/tags`)
ordered by last_updated confirms each tag below is the newest `-aarch64` tag in its
series (2024.2.11.0-b36 pushed 2026-08-25; 2025.1.4.0-b103 2026-04-20; 2025.2.5.2-b5
2026-08-03; 2026.1.1.1-b2 2026-08-18; 2.25.2.0-b359 2025-05-20) and that zero
2.27/2.29/2.31 tags exist (v2.25 = last published preview series). Tag SHAs dereferenced
via GitHub API (`git/ref/tags` + `git/tags`); master HEAD via `commits/master` =
77ce2ba47bd2b84265a7f65650f476496f7056fd (2026-09-01T13:19:13Z).

| target | latest resolved version | source revision | observed repro/control | effective status |
|---|---|---|---|---|
| 2024.2 (LTS) | 2024.2.11.0-b36 (`yugabytedb/yugabyte:2024.2.11.0-b36-aarch64`) | v2024.2.11.0 = 04544fd94fb4e50d05994784f444b075a4a5dfdb | default INSERT -> `PACKED_ROW[0]`, no `SystemColumnId(0)`; flag=false INSERT -> `SystemColumnId(0) -> null` + per-column KV | delivered (default-on); residual liveness only when packing disabled |
| 2025.1 (STS) | 2025.1.4.0-b103 (`...:2025.1.4.0-b103-aarch64`) | v2025.1.4.0 = 690ac4141e57d060fd834312f396643e8c41f260 | same both cells | delivered (default-on); same residual |
| 2025.2 (LTS) | 2025.2.5.2-b5 (`...:2025.2.5.2-b5-aarch64`) | v2025.2.5.2 = 84b2d0064618859f160408c0ce1cb33985581943 | same both cells (`PACKED_ROW_V1[0]`) | delivered (default-on); same residual |
| 2026.1 (STS) | 2026.1.1.1-b2 (`...:2026.1.1.1-b2-aarch64`) | v2026.1.1.1 = 3b010e021f6027744bf9e016b17e108a2eb24e44 | same both cells (`PACKED_ROW_V1[0]`) | delivered (default-on); same residual |
| master | preview image 2.25.2.0-b359 (`...:2.25.2.0-b359-aarch64`) + master HEAD source check | v2.25.2.0 = 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d (image); HEAD 77ce2ba47bd2b84265a7f65650f476496f7056fd (source) | image: same both cells (`PACKED_ROW[0]`). HEAD source: identical mechanism — packed default `pgsql_operation.cc:115-119`, liveness only in unpacked fallback `:1633-1637` | delivered (default-on); same residual |

Reported release: none — the issue (2020-05-01) predates the liveness-column removal and
names no version; nothing distinct to test.

First-delivered provenance (tag/commit ancestry, NOT runtime proof — runtime proof is
the matrix above):
- Packed-row write path + `ysql_enable_packed_row` flag machinery: 59006138e5
  ("[#3520] DocDB: Packed row configuration flags", 2022-06-08) and related #3520 work
  (2.15.x preview lineage).
- Default-on for new universes: v2.20.0.0 (tag SHA 0026607ed49516b4d5770f5479dd5d60d44710af)
  carries `DEFINE_RUNTIME_AUTO_bool(ysql_enable_packed_row, kNewInstallsOnly, false, true)`
  in release builds (its `src/yb/docdb/pgsql_operation.cc:97-101`), matching the public
  doc statement "Packed row for YSQL is enabled by default for new universes in v2.20.0
  and later" (`docs/content/stable/architecture/docdb/packed-rows.md:34`).
- Every in-support branch (2024.2, 2025.1, 2025.2, 2026.1) and the 2.25 preview series
  descends from v2.20 -> the delivered behavior is present everywhere; no backports
  required.

## Proposed fix direction
No code fix warranted. Options for maintainers:
1. Close #4364 as delivered-by-packed-rows (option 2 of the issue, verbatim), noting the
   residual: the unpacked path (`pgsql_operation.cc:1633-1637`) still writes the liveness
   column unconditionally, which only matters for universes running
   `ysql_enable_packed_row=false`.
2. If the residual is ever worth pursuing, the change would be in
   `PgsqlWriteOperation::ApplyInsert` to skip the liveness write when the schema has at
   least one non-nullable non-PK column — but it inherits the issue's own caveat (ALTER
   TABLE making all columns nullable must resume liveness writes) plus read-path
   assumptions (`doc_reader.cc` row-existence handling) and compaction interplay; poor
   cost/benefit for a non-default path.
3. Cosmetic: the `kLivenessColumn` enum comment (`src/yb/dockv/key_entry_value.h:46`)
   still describes only the TTL use, not row existence — could be fixed alongside docs
   (`docs/content/stable/architecture/docdb/data-model.md:229` describes liveness under
   YCQL TTL only).

## Test strategy
Not applicable for a code fix (none proposed). The delivered behavior is already covered
by packed-row tests (`src/yb/docdb/docdb_pgapi*`, packed-row unit tests under
`src/yb/dockv/` and `src/yb/docdb/`, e.g. `docdb_compaction_context` repack tests). The
container demo above (`/tmp/claude/yb4364-repro.sh`) is rerunnable by a maintainer via
public entry points only (ysqlsh, yb-ts-cli, yb-admin, sst_dump).

## Confidentiality check
Artifact reviewed: no customer data, PII, internal hostnames/URLs, secrets, or
non-public information. All identifiers are synthetic (`t_pack`, `t_nopack`, k/v),
all addresses are loopback, all versions/images/commits/docs are public. The one
build-path string visible in logs (`/share/jenkins/workspace/...`) comes from the public
Docker image's own --help output and appears only in local logs, not in any public
artifact.

**Comment by kriteshranjan** [comment]
AI validation notes (`yb4364-repro.sh.txt`):

#!/bin/bash
# yugabyte/yugabyte-db#4364 behavior-delta demo.
# Cell A (control, default flags): YSQL INSERT writes ONE packed top-of-row
#   record at the row DocKey (no subkey) -- no liveness system column.
# Cell B (legacy path, ysql_enable_packed_row=false): YSQL INSERT writes the
#   liveness column SystemColumnId(0) + one KV per column, even though the
#   table has a non-nullable non-PK column.
# Usage: yb4364-repro.sh <short-name> <image>
set -u
NAME="$1"; IMAGE="$2"
CTR="yb4364-$NAME"
LOG="/tmp/claude/yb4364-matrix-$NAME.log"
exec > >(tee "$LOG") 2>&1

echo "### target=$NAME image=$IMAGE $(date -u 2>/dev/null || true)"
podman rm -f "$CTR" >/dev/null 2>&1
podman run -d --name "$CTR" "$IMAGE" bash -c \
  '/home/yugabyte/bin/yugabyted start --advertise_address 127.0.0.1 --base_dir=/tmp/ybd --daemon=false' >/dev/null || exit 1

# Wait for YSQL to accept connections.
ok=""
for i in $(seq 1 60); do
  if podman exec "$CTR" bash -c '/home/yugabyte/bin/ysqlsh -h 127.0.0.1 -U yugabyte -d yugabyte -c "SELECT 1" >/dev/null 2>&1'; then ok=1; break; fi
  sleep 5
done
[ -z "$ok" ] && { echo "RESULT $NAME: YSQL_NEVER_READY"; exit 1; }
echo "--- ysql ready after ~$((i*5))s"

podman exec "$CTR" bash -c '/home/yugabyte/bin/ysqlsh -h 127.0.0.1 -U yugabyte -d yugabyte \
  -c "CREATE TABLE t_pack   (k INT PRIMARY KEY, v INT NOT NULL) SPLIT INTO 1 TABLETS" \
  -c "CREATE TABLE t_nopack (k INT PRIMARY KEY, v INT NOT NULL) SPLIT INTO 1 TABLETS"' || exit 1

echo "--- effective ysql_enable_packed_row before cell A:"
podman exec "$CTR" bash -c 'curl -s "http://127.0.0.1:9000/api/v1/varz" | python3 -c "
import sys,json
for f in json.load(sys.stdin)[\"flags\"]:
    if f[\"name\"]==\"ysql_enable_packed_row\": print(f[\"name\"],\"=\",f[\"value\"],\"(\",f[\"type\"],\")\")
"' || true

# Cell A: packed default.
podman exec "$CTR" bash -c '/home/yugabyte/bin/ysqlsh -h 127.0.0.1 -U yugabyte -d yugabyte -c "INSERT INTO t_pack VALUES (1, 42)"' || exit 1

# Flip packing off at runtime (tserver-side write path decision).
podman exec "$CTR" bash -c '/home/yugabyte/bin/yb-ts-cli --server_address=127.0.0.1:9100 set_flag -force ysql_enable_packed_row false' || exit 1
echo "--- effective ysql_enable_packed_row after set_flag:"
podman exec "$CTR" bash -c 'curl -s "http://127.0.0.1:9000/api/v1/varz" | python3 -c "
import sys,json
for f in json.load(sys.stdin)[\"flags\"]:
    if f[\"name\"]==\"ysql_enable_packed_row\": print(f[\"name\"],\"=\",f[\"value\"],\"(\",f[\"type\"],\")\")
"' || true

# Cell B: unpacked legacy path.
podman exec "$CTR" bash -c '/home/yugabyte/bin/ysqlsh -h 127.0.0.1 -U yugabyte -d yugabyte -c "INSERT INTO t_nopack VALUES (1, 42)"' || exit 1

# Flush both tables to SSTs.
podman exec "$CTR" bash -c '/home/yugabyte/bin/yb-admin --master_addresses=127.0.0.1:7100 flush_table ysql.yugabyte t_pack 60 && /home/yugabyte/bin/yb-admin --master_addresses=127.0.0.1:7100 flush_table ysql.yugabyte t_nopack 60' || exit 1
sleep 3

# Map table name -> table id, find the tablet dir, sst_dump it.
for tbl in t_pack t_nopack; do
  echo "=== sst_dump $NAME/$tbl ==="
  podman exec -e TBL="$tbl" "$CTR" bash -c '
    tid=$(/home/yugabyte/bin/yb-admin --master_addresses=127.0.0.1:7100 list_tables include_table_id | awk -v t="yugabyte.$TBL" "\$1==t {print \$NF}")
    echo "table_id=$tid"
    d=$(find /tmp/ybd -type d -name "table-$tid" | grep -v intents | head -1)
    tabletdir=$(find "$d" -maxdepth 1 -type d -name "tablet-*" | grep -v snapshots | head -1)
    echo "tabletdir=$tabletdir"
    ls "$tabletdir" | grep -c sst >/dev/null || echo "NO_SST_FILES"
    for sst in "$tabletdir"/*.sst; do
      [ -e "$sst" ] || continue
      /home/yugabyte/bin/sst_dump --command=scan --file="$sst" --output_format=decoded_regulardb 2>/dev/null | grep -v "^Process\|^Sst file format" 
    done'
done

podman rm -f "$CTR" >/dev/null 2>&1
echo "### done $NAME"

**Comment by kriteshranjan** [comment]
AI validation notes (`yb4364-matrix.log.txt`):

### target=2024.2 image=docker.io/yugabytedb/yugabyte:2024.2.11.0-b36-aarch64 Tue  1 Sep 2026 14:27:05 UTC
--- ysql ready after ~15s
CREATE TABLE
CREATE TABLE
--- effective ysql_enable_packed_row before cell A:
ysql_enable_packed_row = true ( Auto )
INSERT 0 1
--- effective ysql_enable_packed_row after set_flag:
ysql_enable_packed_row = false ( Custom )
INSERT 0 1
Flushed [yugabyte.t_pack] tables.
Flushed [yugabyte.t_nopack] tables.
=== sst_dump 2024.2/t_pack ===
table_id=000033cf000030008000000000004000
tabletdir=/tmp/ybd/data/yb-data/tserver/data/rocksdb/table-000033cf000030008000000000004000/tablet-f7518b228df649e790d52ddff527bd30
from [] to []
SubDocKey(DocKey(0x1210, [1], []), [HT{ physical: 1788272846558609 }]) -> PACKED_ROW[0](05000000480000002A)
=== sst_dump 2024.2/t_nopack ===
table_id=000033cf000030008000000000004005
tabletdir=/tmp/ybd/data/yb-data/tserver/data/rocksdb/table-000033cf000030008000000000004005/tablet-a8df03f9b388442bbf7ee2fdbf336129
from [] to []
SubDocKey(DocKey(0x1210, [1], []), [SystemColumnId(0); HT{ physical: 1788272846992584 }]) -> null
SubDocKey(DocKey(0x1210, [1], []), [ColumnId(1); HT{ physical: 1788272846992584 w: 1 }]) -> 42
### done 2024.2
### target=2025.1 image=docker.io/yugabytedb/yugabyte:2025.1.4.0-b103-aarch64 Tue  1 Sep 2026 14:26:34 UTC
--- ysql ready after ~15s
CREATE TABLE
CREATE TABLE
--- effective ysql_enable_packed_row before cell A:
ysql_enable_packed_row = true ( Auto )
INSERT 0 1
--- effective ysql_enable_packed_row after set_flag:
ysql_enable_packed_row = false ( Custom )
INSERT 0 1
Flushed [yugabyte.t_pack [ysql_schema=public] [000034cb000030008000000000004000]] tables.
Flushed [yugabyte.t_nopack [ysql_schema=public] [000034cb000030008000000000004005]] tables.
=== sst_dump 2025.1/t_pack ===
table_id=000034cb000030008000000000004000
tabletdir=/tmp/ybd/data/yb-data/tserver/data/rocksdb/table-000034cb000030008000000000004000/tablet-273bd7927e1248748fd50fa1db97fc53
from [] to []
SubDocKey(DocKey(0x1210, [1], []), [HT{ physical: 1788272815494229 }]) -> PACKED_ROW[0](05000000480000002A)
=== sst_dump 2025.1/t_nopack ===
table_id=000034cb000030008000000000004005
tabletdir=/tmp/ybd/data/yb-data/tserver/data/rocksdb/table-000034cb000030008000000000004005/tablet-d8db7e07ef3a4bf380b5b613953f01ab
from [] to []
SubDocKey(DocKey(0x1210, [1], []), [SystemColumnId(0); HT{ physical: 1788272815907762 }]) -> null
SubDocKey(DocKey(0x1210, [1], []), [ColumnId(1); HT{ physical: 1788272815907762 w: 1 }]) -> 42
### done 2025.1
### target=2025.2 image=docker.io/yugabytedb/yugabyte:2025.2.5.2-b5-aarch64 Tue  1 Sep 2026 14:26:02 UTC
--- ysql ready after ~15s
CREATE TABLE
CREATE TABLE
--- effective ysql_enable_packed_row before cell A:
ysql_enable_packed_row = true ( Auto )
INSERT 0 1
--- effective ysql_enable_packed_row after set_flag:
ysql_enable_packed_row = false ( Custom )
INSERT 0 1
Flushed [yugabyte.t_pack [ysql_schema=public] [000034e1000030008000000000004000]] tables.
Flushed [yugabyte.t_nopack [ysql_schema=public] [000034e1000030008000000000004005]] tables.
=== sst_dump 2025.2/t_pack ===
table_id=000034e1000030008000000000004000
tabletdir=/tmp/ybd/data/yb-data/tserver/data/rocksdb/table-000034e1000030008000000000004000/tablet-9b6fa78c446e444683555f621c9e6c89
from [] to []
SubDocKey(DocKey(0x1210, [1], []), [HT{ physical: 1788272783749891 }]) -> PACKED_ROW_V1[0](05000000480000002A)
=== sst_dump 2025.2/t_nopack ===
table_id=000034e1000030008000000000004005
tabletdir=/tmp/ybd/data/yb-data/tserver/data/rocksdb/table-000034e1000030008000000000004005/tablet-71691ce1aae24832999acbd28ced6ead
from [] to []
SubDocKey(DocKey(0x1210, [1], []), [SystemColumnId(0); HT{ physical: 1788272784154490 }]) -> null
SubDocKey(DocKey(0x1210, [1], []), [ColumnId(1); HT{ physical: 1788272784154490 w: 1 }]) -> 42
### done 2025.2
### target=2026.1 image=docker.io/yugabytedb/yugabyte:2026.1.1.1-b2-aarch64 Tue  1 Sep 2026 14:25:25 UTC
--- ysql ready after ~15s
CREATE TABLE
CREATE TABLE
--- effective ysql_enable_packed_row before cell A:
ysql_enable_packed_row = true ( Auto )
INSERT 0 1
--- effective ysql_enable_packed_row after set_flag:
ysql_enable_packed_row = false ( Custom )
INSERT 0 1
Flushed [yugabyte.t_pack [ysql_schema=public] [000034e1000030008000000000004000]] tables.
Flushed [yugabyte.t_nopack [ysql_schema=public] [000034e1000030008000000000004005]] tables.
=== sst_dump 2026.1/t_pack ===
table_id=000034e1000030008000000000004000
tabletdir=/tmp/ybd/data/yb-data/tserver/data/rocksdb/table-000034e1000030008000000000004000/tablet-323eb28ba1684da3970ef226a2b53e19
from [] to []
SubDocKey(DocKey(0x1210, [1], []), [HT{ physical: 1788272746494057 }]) -> PACKED_ROW_V1[0](05000000480000002A)
=== sst_dump 2026.1/t_nopack ===
table_id=000034e1000030008000000000004005
tabletdir=/tmp/ybd/data/yb-data/tserver/data/rocksdb/table-000034e1000030008000000000004005/tablet-4a1542bc1c2f4da38021f170be49a4cc
from [] to []
SubDocKey(DocKey(0x1210, [1], []), [SystemColumnId(0); HT{ physical: 1788272746885627 }]) -> null
SubDocKey(DocKey(0x1210, [1], []), [ColumnId(1); HT{ physical: 1788272746885627 w: 1 }]) -> 42
### done 2026.1
### target=2.25.2 image=docker.io/yugabytedb/yugabyte:2.25.2.0-b359-aarch64 Tue  1 Sep 2026 14:27:38 UTC
--- ysql ready after ~15s
CREATE TABLE
CREATE TABLE
--- effective ysql_enable_packed_row before cell A:
ysql_enable_packed_row = true ( Auto )
INSERT 0 1
--- effective ysql_enable_packed_row after set_flag:
ysql_enable_packed_row = false ( Custom )
INSERT 0 1
Flushed [yugabyte.t_pack [ysql_schema=public] [000034cb000030008000000000004000]] tables.
Flushed [yugabyte.t_nopack [ysql_schema=public] [000034cb000030008000000000004005]] tables.
=== sst_dump 2.25.2/t_pack ===
table_id=000034cb000030008000000000004000
tabletdir=/tmp/ybd/data/yb-data/tserver/data/rocksdb/table-000034cb000030008000000000004000/tablet-1b89faa0e94a4b9287563b7cbfc09f38
from [] to []
SubDocKey(DocKey(0x1210, [1], []), [HT{ physical: 1788272878644294 }]) -> PACKED_ROW[0](05000000480000002A)
=== sst_dump 2.25.2/t_nopack ===
table_id=000034cb000030008000000000004005
tabletdir=/tmp/ybd/data/yb-data/tserver/data/rocksdb/table-000034cb000030008000000000004005/tablet-147b503b178a404e9ac92260c7f54ca5
from [] to []
SubDocKey(DocKey(0x1210, [1], []), [SystemColumnId(0); HT{ physical: 1788272879021294 }]) -> null
SubDocKey(DocKey(0x1210, [1], []), [ColumnId(1); HT{ physical: 1788272879021294 w: 1 }]) -> 42
### done 2.25.2

**Comment by kriteshranjan** [comment]
Closing this issue based on AI validation. cc: @m-iancu , @VamsidharGunturu
